features/steps/jobs_steps.go

```diff
@@ -121,22 +120,29 @@ func (f *JobsFeature) RegisterSteps(ctx *godog.ScenarioContext) {
 }
 
 // Reset sets the resources within a specific JobsFeature back to their default values.
-func (f *JobsFeature) Reset(mongoFail bool) *JobsFeature {
+func (f *JobsFeature) Reset(mongoFail bool) error {
+	var err error
 	if mongoFail {
 		f.MongoClient.Database = "lost database connection"
 	} else {
 		f.MongoClient.Database = memongo.RandomDatabase()
 	}
 	ctx := context.Background()
-	cfg, err := config.Get()
-	if err != nil {
-		return nil
+	if f.Config == nil {
+		cfg, err := config.Get()
+		if err != nil {
+			return err
+		}
+		f.Config = cfg
 	}
-	err = f.MongoClient.Init(ctx, cfg)
-	if err != nil {
-		return nil
+	if f.MongoClient == nil {
+		err = f.MongoClient.Init(ctx, f.Config)
```

Comment:
This doesn't look right. If `f.MongoClient == nil` this would panic when trying to call a function on it?

Reply:
I added the 'if f.MongoClient == nil' statement because otherwise the component tests produce the following error: failed to initialise Mongo client: session already exists

Reply:
Still, you can't call a function on a nil pointer - it will panic. What it looks like is that you need to remove the re-initialisation from this function altogether.

Reply:
I see what you mean, thanks, I've removed that now